state/dashboard/operations/searchMovies.js

```diff
@@ -3,34 +3,37 @@ import { normalize, schema } from 'normalizr';
 
 import { SEARCH_MOVIES } from '../types';
 import { mergeMoviesList } from '../../data/actions';
-import { fetchTrendingMoviesSuccess } from '../actions';
+import {
+  fetchMoviesSuccess,
+  fetchMoviesError,
+  setSearchMoviesQuery
+} from '../actions';
 
 const searchMoviesLogic = createLogic({
   type: SEARCH_MOVIES,
   warnTimeout: 0,
 
-  async process({ httpClient, action }, dispatch, done) {
-    const {
-      search,
-      setErrors,
-      setSubmitting,
-      page
-    } = action;
+  async process({ getState, httpClient, action }, dispatch, done) {
+    const { searchQuery, page } = action;
     const currentPage = page;
+    const query = searchQuery || getState().dashboard.searchQuery;
```

Comment:
Use selector to get values from state

Reply:
Now we don't need to take searchQuery from the state.
searchMovies action always has a query.